src/Bicep.LangServer/Deploy/DeploymentHelper.cs

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.IO;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Azure.ResourceManager;
+using Azure.ResourceManager.Resources;
+using Azure.ResourceManager.Resources.Models;
+using Bicep.Core;
+
+namespace Bicep.LanguageServer.Deploy
+{
+    public class DeploymentHelper
+    {
+        public static async Task<string> CreateDeployment(
+            IDeploymentCollectionProvider deploymentCollectionProvider,
+            ArmClient armClient,
+            string template,
+            string parameterFilePath,
+            string id,
+            string scope,
+            string location)
+        {
+            if ((scope == LanguageConstants.TargetScopeTypeSubscription ||
+                scope == LanguageConstants.TargetScopeTypeManagementGroup) &&
+                string.IsNullOrWhiteSpace(location))
+            {
+                return LangServerResources.MissingLocationDeploymentFailedMessage;
+            }
+
+            DeploymentCollection? deploymentCollection;
+            var resourceIdentifier = new ResourceIdentifier(id);
+
+            if (scope == LanguageConstants.TargetScopeTypeResourceGroup ||
+                scope == LanguageConstants.TargetScopeTypeSubscription ||
+                scope == LanguageConstants.TargetScopeTypeManagementGroup)
+            {
+                try
+                {
+                    deploymentCollection = deploymentCollectionProvider.GetDeploymentCollection(armClient, resourceIdentifier, scope);
+                }
+                catch (Exception e)
+                {
+                    return string.Format(LangServerResources.DeploymentFailedWithExceptionMessage, e.Message);
+                }
+            }
+            else
+            {
+                return LangServerResources.UnsupportedScopeDeploymentFailedMessage;
+            }
+
+            if (deploymentCollection is not null)
+            {
+                JsonElement parameters;
+
+                if (string.IsNullOrWhiteSpace(parameterFilePath))
+                {
+                    parameters = JsonDocument.Parse("{}").RootElement;
```

Comment:
Can you use one of the `JsonElementFactory.CreateElement` helper methods instead? `JsonDocument.Parse` returns an instance of `JsonDocument` which implements `IDisposable`, so `JsonDocument.Parse` must be called in a using statement. The `JsonElementFactory.CreateElement` helper methods already handle that.

Reply:
Same for the `JsonDocument.Parse` calls below.

Reply:
Updated here:
Sure, updated as part of this commit:
https://github.com/Azure/bicep/commit/32bed56af070ae3b2c2b494c956f07cf68874667

Thanks!